src/config_file.c

```diff
@@ -1720,9 +1720,48 @@ static int parse_multiline_variable(struct reader *reader, git_buf *value, int i
 	return 0;
 }
 
+GIT_INLINE(bool) is_namechar(char c)
+{
+	return isalnum(c) || c == '-';
+}
+
+static int parse_name(
+	char **name, const char **value, struct reader *reader, const char *line)
+{
+	const char *c, *name_end = NULL, *value_start;
+
+	*name = NULL;
+	*value = NULL;
+
+	for (c = line; *c && is_namechar(*c); c++)
+		;
```

Comment:
We have two different ways of applying a function to a sequence of chars, one here and the while for `git__isspace()`. The other one does not lead to an empty loop body, could we change this to the style of the `while` below?